packages/ketcher-core/src/domain/AttachmentPoint.ts

```diff
@@ -0,0 +1,358 @@
+      .attr('stroke', 'white')
+      .attr('stroke-width', '1px')
+      .attr('fill', fill);
+
+    const labelGroup = attachmentPoint.append('text');
+
+    labelGroup
+      .text(this.RN)
+      .attr('x', labelCoordinatesOnMonomer.x)
+      .attr('y', labelCoordinatesOnMonomer.y)
+      .style('font-size', '12px')
+      .style('fill', '#585858')
+      .style('user-select', 'none');
+
+    return attachmentPoint;
+  }
+
+  public appendAttachmentPointToBond() {
+    let rad;
+
+    if (
+      this.monomer.id ===
+      this.monomer.attachmentPointsToBonds[this.RN]?.firstMonomer?.id
+    ) {
+      rad = this.rotateToAngle(
+        this.monomer.attachmentPointsToBonds[this.RN],
+        true,
+      );
+    } else {
+      rad = this.rotateToAngle(this.monomer.attachmentPointsToBonds[this.RN]);
+    }
+    const deg = Vec2.radians_to_degrees(rad);
+
+    const [pointOnBorder, pointOfAttachment, labelPoint] =
+      this.catchThePoint(deg);
+    const attachmentOnBorder =
+      this.canvas_to_monomer_coordinates(pointOnBorder);
+
+    const attachmentPointCoordinates =
+      this.canvas_to_monomer_coordinates(pointOfAttachment);
+
+    const labelCoordinatesOnMonomer =
+      this.canvas_to_monomer_coordinates(labelPoint);
+
+    const attachmentPoint = this.renderUsedAttachmentPointByCoordinates(
+      attachmentOnBorder,
+      attachmentPointCoordinates,
+      labelCoordinatesOnMonomer,
+    );
+
+    this.element = attachmentPoint;
+    return attachmentPoint;
+  }
+
+  public rotateToAngle(polymerBond: PolymerBond | null, flip = false) {
+    if (!polymerBond) {
+      return 0;
+    }
+    let rad;
+    if (flip) {
+      rad = Vec2.oxAngleForVector(
+        polymerBond.endPosition,
+        polymerBond.position,
+      );
+    } else {
+      rad = Vec2.oxAngleForVector(
+        polymerBond.position,
+        polymerBond.endPosition,
+      );
+    }
+
+    return rad;
+  }
+
+  private catchThePoint(rotationAngle: number): { x: number; y: number }[] {
+    if (!this.monomer.renderer) return Array(3).fill({ x: 0, y: 0 });
+
+    const lengthInitial = (this.bodyWidth + this.bodyHeight) / 2.3;
+
+    const currentMonomerCenter = {
+      x: this.monomer.renderer.center.x,
+      y: this.monomer.renderer.center.y,
+    };
+
+    this.initialAngle = rotationAngle - 180;
+
+    const pointOnBorder = this.fastFindThePoint(
+      currentMonomerCenter,
+      lengthInitial,
+    );
+
+    const [labelPoint, pointOfAttachment] = this.findLabelPoint(
+      pointOnBorder,
+      this.initialAngle,
+    );
+    return [pointOnBorder, pointOfAttachment, labelPoint];
+  }
+
+  private fastFindThePoint(coordStart, length, angle = this.initialAngle) {
+    const angleRadians = Vec2.degrees_to_radians(angle);
+
+    const cos = Math.cos(angleRadians);
+    const sin = Math.sin(angleRadians);
+
+    const deltaX = length * cos;
+    const deltaY = length * sin;
+
+    const x2 = coordStart.x + deltaX;
+    const y2 = coordStart.y + deltaY;
```

Comment:
I believe we can add this code to Vec2. It is an operation of addition the vector(represented as point, line length and angle) to point. Probably we can name it something like "findPointOnVectorByLength"